frontend/components/Blocks/HeroBlock/HeroBlock.tsx

```diff
@@ -0,0 +1,117 @@
+
+type Props = {
+  data: {
+    type: string;
+    value: {
+      backgroundColor: string;
+      title: string;
+      text: string;
+      media: [];
+      alt_text: string;
+      button?: {
+        button_style: string;
+        button_text: string;
+        button_hyperlink: string;
+      };
+    };
+    id: string;
+  };
+};
+
+export default function HeroBlock({ data }: Props) {
+  const [imageSize, setImageSize] = useState({
+    width: 1,
+    height: 1,
+  });
+  const [hasWindow, setHasWindow] = useState(false);
+
+  const backgroundcolor = data.value.backgroundColor;
+  const mediaDetails = data.value.media[0];
+
+  useEffect(() => {
+    if (typeof window !== "undefined") {
+      setHasWindow(true);
+    }
+  }, []);
+
+  const scrollDown = () => {
+    const viewportHeight = window.innerHeight;
+    window.scrollTo({
+      top: viewportHeight,
+    });
+  };
+
+  return (
+    <div className={``}>
+      <div className={`flex flex-col justify-center w-full ${backgroundcolor}`}>
+        <div className={`flex flex-col  lg:flex-row `}>
+          <div className="flex flex-col lg:w-1/2 py-12 px-10 lg:px-16">
+            <h1>
+              <RawHtml html={data.value.title}></RawHtml>
+            </h1>
+            <div className={`text-3xl font-normal mt-8 mr-8`}>
+              <RawHtml html={data.value.text}></RawHtml>
+            </div>
+          </div>
+
+          <div className="flex flex-col py-12 px-10 lg:px-16 lg:pt-24 lg:w-1/2">
+            {(() => {
+              let returnValue = "";
+              switch (mediaDetails.type) {
+                case "video":
+                  returnValue =
+                    mediaDetails.value && hasWindow ? (
+                      <ReactPlayer
+                        key={`player ${mediaDetails.value.id}`}
+                        url={mediaDetails.value}
+                      />
+                    ) : (
+                      ""
+                    );
+                  break;
+                case "image":
+                  returnValue = mediaDetails.value ? (
+                    <Image
+                      src={process.env.NEXT_PUBLIC_BASE_URL + "/" + mediaDetails.value.img.src}
+                      alt={mediaDetails.value.img.alt}
+                      layout="responsive"
+                      objectFit="contain"
+                      priority={true}
+                      onLoadingComplete={target => {
+                        setImageSize({
+                          width: target.naturalWidth,
+                          height: target.naturalHeight,
+                        });
+                      }}
+                      width={imageSize.width}
+                      height={imageSize.height}
+                      className="image"
+                    />
+                  ) : (
+                    ""
+                  );
+                default:
+              }
+
+              return <div>{returnValue}</div>;
+            })()}
```

Comment:
I find this whole block is quite confusing. I'm not sure why the `<div>`'s child is an immediately-invoked function.

Could this be extracted higher in the component, or perhaps to a separate component?

```tsx
function MediaContent(...) {
  case "video": {
    // ...
  }

  case: "image": {
    // ...
  }
}

<MediaContent mediaDetails={mediaDetails} />
```

Reply:
Thanks good point. I've extracted it to a seperate component which can be used in both the HeroBlock and the TextMedia block.